### PR TITLE
planner: misuse like2regexp and regexp when to query information.schema

### LINKED ISSUE #32451: INFORMATION_SCHEMA.COLUMNS handles SQL like wild cards wrong
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

select table_schema, table_name, column_name from information_schema.columns where column_name like '%time';

I would assume this is similar for other information_schema tables and possibly other generated tables in TiDB.

### 2. What did you expect to see? (Required)

Column names only ending with 'time'

### 3. What did you see instead (Required)

All columns matching the regex pattern `(?i).*time` (without an ending '$')
Including column_name 'DATETIME_PRECISION', see below:

```
tidb> select table_schema, table_name, column_name from information_schema.columns where column_name like '%time';
+--------------------+---------------------------------------------------+-----------------------------------------+
| table_schema       | table_name                                        | column_name                             |
+--------------------+---------------------------------------------------+-----------------------------------------+
| INFORMATION_SCHEMA | STATEMENTS_SUMMARY_EVICTED                        | BEGIN_TIME                              |
| INFORMATION_SCHEMA | STATEMENTS_SUMMARY_EVICTED                        | END_TIME                                |
| INFORMATION_SCHEMA | DEADLOCKS                                         | OCCUR_TIME                              |
| INFORMATION_SCHEMA | PARAMETERS                                        | DATETIME_PRECISION                      |
```

```
tidb> explain select table_schema, table_name, column_name from information_schema.columns where column_name like '%time';
+----------------------+----------+------+---------------+----------------------------------+
| id                   | estRows  | task | access object | operator info                    |
+----------------------+----------+------+---------------+----------------------------------+
| Projection_4         | 10000.00 | root |               | Column#2, Column#3, Column#4     |
| └─MemTableScan_5     | 10000.00 | root | table:COLUMNS | column_name_pattern:[(?i).*time] |
+----------------------+----------+------+---------------+----------------------------------+
2 rows in set (0.01 sec)
```
### 4. What is your TiDB version? (Required)

```
Release Version: v5.5.0-alpha-288-g64f8c4756f
Edition: Community
Git Commit Hash: 64f8c4756fbb0bc48683a707ddca25b7560f0719
Git Branch: master
UTC Build Time: 2022-02-18 00:22:36
GoVersion: go1.17.6
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #32451

Problem Summary:

Previously, a push-down was added to fix a case-sensitive issue. but a bug is imported that it misuses the like2repexp and regexp to implement ```like```. it causes inaccurate search results

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by zimulala** [comment]
Please add a description, so far just write `close # XXX`. Also, please remember to fill in Release Note.

**Review by mjonss — APPROVED** [review]
LGTM

**Comment by hawkingrei** [comment]
> Please add a description, so far just write `close # XXX`. Also, please remember to fill in Release Note.

Have added a description.

File: planner/core/stringer_test.go [hunk 1]
```diff
@@ -66,11 +66,11 @@ func TestPlanStringer(t *testing.T) {
 		},
 		{
 			sql:  "show tables in test like 't%'",
-			plan: "Show(table_pattern:[(?i)t.*])",
+			plan: "Show(table_pattern:[t%])",
 		},
 		{
 			sql:  "show tables in test like '%T%'",
-			plan: "Show(table_pattern:[(?i).*T.*])",
+			plan: "Show(table_pattern:[%t%])",
```
Comment (Defined2014) [inline planner/core/stringer_test.go:73]: Not sure is this a good way. We can't get `case-insensitive` information from the plan right now.

**Review by Defined2014 — COMMENTED** [review]
None of us have `write` permits for TiDB repo. Need one more committer or maintainer for this PR.

File: planner/core/stringer_test.go (hunk 1 above)
Comment (hawkingrei) [reply planner/core/stringer_test.go:73]: In fact, We need case-insensitivity when to ```show``` executor. so we add case-insensitive push-down to the ```show```.

**Review by Defined2014 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**